### ISSUE TITLE
Warining: "listener not found" in L.imageOverlay@18.0.0

### ISSUE THREAD

**Issue by sprout2000** [Leaflet/Leaflet#8166]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

After updating to [v18](https://github.com/Leaflet/Leaflet/releases/tag/v1.8.0), I suddenly started receiving the following warning in `L.imageOverlay`:

```sh
▶️ listener not found    Events.js:180

_off @ Events.js:180
off @ Events.js:89
setMaxBounds @ Map.js:453
initialize @ Map.js:145
i @ Class.js:24
t.map @ Map.js:1748
img.onload @ index.html:30
load (async)
(anonymous) @ index.html:25
```

**Steps to reproduce**

```html
<div id="map"></div>

<script>
  const img = new Image();

  img.onload = () => {
    const bounds = new L.LatLngBounds([
      [img.height, 0],
      [0, img.width],
    ]);

    const map = L.map('map', {
      crs: L.CRS.Simple,
      maxBounds: bounds,
    }).fitBounds(bounds);

    L.imageOverlay(img.src, bounds).addTo(map);
  };

  img.src = './image.jpg';
</script>
```

**Expected behavior**

What does this warning mean?

**Environment**
- Leaflet version: 18.0.0
- Browser (with version): Google Chrome 100.0.4896.127
- OS/Platform (with version): macOS 12.3.1 (Monterey)

**Minimal example reproducing the issue**

https://codesandbox.io/s/condescending-kirch-hq7tbz

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

**Comment by Falke-Design** [comment]
Thank you for reporting. We have already a fix for this: #8161

Duplicate of #7951

**Comment by sprout2000** [comment]
Thank you very much.
Sorry for opening duplicate issue.
